lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -513,7 +513,11 @@ def test_hash_resource(self):
         decoded or otherwise.
         """
         resources = ['ASCII text', u'❄ I am a special snowflake.', "❄ So am I, but I didn't tell you."]
-        self.assertEqual(hash_resource(resources), 'a76e27c8e80ca3efd7ce743093aa59e0')
+        if six.PY2:
+            excepted_hash = 'a76e27c8e80ca3efd7ce743093aa59e0'
+        else:
+            excepted_hash = 'cbf6900258585588531d2e60e9f9ea96'
+        self.assertEqual(hash_resource(resources), excepted_hash)
```

Comment:
`hash_resource` is defined like this:
```
def hash_resource(resource):
    """
    Hash a :class:`web_fragments.fragment.FragmentResource
    """
    md5 = hashlib.md5()
    for data in resource:
        md5.update(repr(data).encode('utf-8'))
    return md5.hexdigest()
```
We shouldn't be using repr for this, since it varies between versions.  What is this function used for? I hope we don't have to keep hashes consistent across versions, because it will be a pain.

Reply:
BTW, if we do keep this test code, we should spell "expected" properly :)

Reply:
I have fixed the spelling. Lesson: Do not discuss exceptions when writing code. :)